python/ray/tune/tests/test_utils.py

```diff
@@ -88,7 +91,21 @@ def _fail_once():
     assert marker.exists()
 
 
-if __name__ == "__main__":
-    import sys
+def test_tee_recursion():
+    f = io.TextIOBase()
+    g = io.TextIOBase()
+
+    tee = Tee(f, g)
+
+    hdlr = logging.StreamHandler(tee)
+    util_logger.addHandler(hdlr)
+
+    logging.info("BEFORE")
+    f.close()
+    logging.info("AFTER")
```

Comment:
Should `sys.stderr` be one of the streams in order to test the recursion? Or setup a redirect to `g`? Otherwise the `logging.info` after would just emit a `logging.warning` without invoking the Tee again and again -- maybe I'm missing something here!

Reply:
The recursion comes up without the fix. Here we specifically add a loop by adding a streaming handler that logs to `f`. The call to `logging.info` then triggers this handler.

But yes, this should be `util_logger.info()` instead to make it even clearer.

Without the fix, the test runs forever (or eventually runs into the stack recursion problem).